R/filter_occs_time.R

```diff
@@ -0,0 +1,277 @@
+#' filter_occs_time
+#'
+#' @description
+#' A function to trim fossil occurrences to within a specified temporal
+#' duration and/or level of precision. Filtering can be performed using
+#' stratigraphic interval names or absolute ages.
+#'
+#' @param occdf `data.frame`. A `data.frame` of the fossil occurrences you wish
+#' to filter. This `data.frame` must contain at least two columns, maximum
+#' occurrence age and minimum occurrence age (see `max_ma`, `min_ma`), both of
+#' which should be numeric.
+#' @param min_ma `character`. The name of the column you wish to be treated as
+#' the minimum ages in `occdf` and `bins` (defaults to "min_ma").
+#' @param max_ma `character`. The name of the column you wish to be treated as
+#' the maximum ages in `occdf` and `bins` (defaults to "max_ma").
+#' @param bins `data.frame`. A `data.frame` of time bins used to look up
+#' absolute stratigraphic ages for filtering fossil occurrences, such as that
+#' returned by [time_bins()]. This `data.frame` must contain at least
+#' three columns as named by `interval_name` (defaults to "interval_name"),
+#' `max_ma` (defaults to "max_ma") and `min_ma` (defaults to "min_ma"). Columns
+#' `max_ma` and `min_ma` must be numeric.
+#' @param interval_name `character` The name of the column in `bins` that
+#' contains the interval names.
+#' @param youngest_bin `character`. The name of the youngest stratigraphic
+#' interval to trim to. Only one interval name should be supplied, which should
+#' be present in column `interval_name` in `bins`.
+#' @param oldest_bin `character`. The name of the oldest stratigraphic interval
+#' to trim to. Only one interval name should be supplied, which should be
+#' present in column `interval_name` in `bins`.
+#' @param min_age `numeric`. The minimum absolute occurrence age to trim to.
+#' Ignored if `youngest_bin` is supplied.
+#' @param max_age `numeric`. The maximum absolute occurrence age to trim to.
+#' Ignored if `oldest_bin` is supplied.
+#' @param uncertainty A numeric value denoting the maximum tolerable temporal
+#' uncertainty for retaining occurrences. Alternatively, if this argument is set
+#' to "bin", then only occurrences which span no more than `nbin` intervals in
+#' `bins` will be retained.
```

Comment:
I definitely did not expect you to actually implement the `nbin` option, so maybe now this should be "bins" like before 🙈 (I had suggested "bin" because it was originally only one bin)? It probably doesn't matter (and you could also say "bins" but also catch "bin" as well), so I'll leave the decision up to you (and the other reviewer).